File: api/app/enums/status.py
"""
Session status types
"""

from enum import Enum


class StatusEnum(str, Enum):
    """Session status types"""

    created = "created"
    uploaded = "uploaded"
    processing = "processing"
    completed = "completed"
    failed = "failed"

    def __str__(self) -> str:
        return self.value

File: laravel/app/Enums/Status.php
<?php

namespace App\Enums;

enum Status: string
{
    case UPLOADED = 'UPLOADED';
    case PROCESSING = 'PROCESSING';
    case COMPLETED = 'COMPLETED';
    case FAILED = 'FAILED';

    public static function default(): string
    {
        return self::UPLOADED->value;
    }

    /** @return array<int, string> */
    public static function toArray(): array
    {
        return array_column(self::cases(), 'value');
    }
}
